test/core/Leaf.t.sol

```diff
@@ -463,4 +480,53 @@ contract FieldTest is Test {
   function field_isDirty(DirtyField field) public {
     assertEq(field.isDirty(), DirtyField.unwrap(field) & TOPBIT == TOPBIT);
   }
+
+  // Some BitLib.ctz64 relies on specific level sizes
+  function test_level_sizes() public {
+    assertLe(LEVEL3_SIZE, int(MAX_LEVEL_SIZE), "bad level3 size");
+    assertLe(LEVEL2_SIZE, int(MAX_LEVEL_SIZE), "bad level2 size");
+    assertLe(LEVEL1_SIZE, int(MAX_LEVEL_SIZE), "bad level1 size");
+    assertLe(LEVEL0_SIZE, int(MAX_LEVEL_SIZE), "bad level0 size");
+  }
+
+  // checks that there is no overflow
+  function test_maxSafeVolumeIsSafeLowLevel() public {
+    assertGt(MAX_SAFE_VOLUME * ((1 << MANTISSA_BITS) - 1), 0);
+  }
+
+  // non-optimized divExpUp
+  function divExpUp_spec(uint a, uint exp) public returns (uint) {
+    if (exp > 255) return 1;
+    uint den = 2 ** exp;
+    uint carry = a % den == 0 ? 0 : 1;
+    return a / den + carry;
+  }
+
+  function test_inboundFromOutboundUp_and_converse(int logPrice, uint amt) public {
+    amt = bound(amt, 0, MAX_SAFE_VOLUME);
+    logPrice = bound(logPrice, MIN_LOG_PRICE, MAX_LOG_PRICE);
+
+    uint sig;
+    uint exp;
+
+    //inboundFromOutboundUp
+    (sig, exp) = LogPriceConversionLib.nonNormalizedPriceFromLogPrice(logPrice);
+    assertEq(LogPriceLib.inboundFromOutboundUp(logPrice, amt), divExpUp_spec(sig * amt, exp));
+
+    //outboundFromInboundUp
+    (sig, exp) = LogPriceConversionLib.nonNormalizedPriceFromLogPrice(-logPrice);
+    assertEq(LogPriceLib.outboundFromInboundUp(logPrice, amt), divExpUp_spec(sig * amt, exp));
+  }
+
+  function test_outuboundFromInboundUp(int logPrice, uint amt) public {
+    amt = bound(amt, 0, MAX_SAFE_VOLUME);
+    logPrice = bound(logPrice, MIN_LOG_PRICE, MAX_LOG_PRICE);
+
+    (uint sig, uint exp) = LogPriceConversionLib.nonNormalizedPriceFromLogPrice(logPrice);
+    assertEq(LogPriceLib.inboundFromOutboundUp(logPrice, amt), divExpUp_spec(sig * amt, exp));
```

Comment:
Test name indicates this should test `outboundFromInboundUp`?

Reply:
test removed